code/game/objects/items/teleportation.dm

```diff
@@ -427,7 +429,14 @@
 		charges = max(charges - 1, 0)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(current_location)
 		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
-		make_bloods(current_location, destination, user)
+		new /obj/effect/temp_visual/circle_wave/syndi_teleporter(destination)
+		var/blood_ejection = FALSE
+		if(!HAS_TRAIT(user, TRAIT_NOBLOOD))
+			blood_ejection = make_bloods(current_location, destination, user)
+		if(blood_ejection)
```

Comment:
Why not use the return value of `make_bloods`, and check noblood there (if noblood return false)